### DISCUSSION TITLE
Best Training configurations for Faces

### DISCUSSION THREAD

**Discussion by cloneofsimo** [cloneofsimo/lora#43] [Ideas]
My take on the learing rate,
really not anything conclusive, but seems like even higher-learning rate can work as well. (upto 4e-4)

![lora_diff_lrs](https://user-images.githubusercontent.com/35953539/207823943-fc0faa9d-0c7f-493e-a3f4-b37bc0a81d81.jpg)

> prompt = "female 3d game character bnha, Skill magic geek inside matrix deepdream radiating a glowing aura stuff loot legends stylized digital illustration video game icon artstation lois van baarle, ilya kuvshinov, rossdraws", $\alpha = 1.0$

![lora_diff_lrs_0 6](https://user-images.githubusercontent.com/35953539/207823932-7e4c2e5b-bf4f-406c-8efd-666de4283ff5.jpg)

> prompt = "portrait of female 3d game character bnha, impressionist style from the 19th century, claude monet, oil painting", $\alpha = 0.6$

All with  `num_inference_steps=50, guidance_scale=4.5`, Euler A scheduler

Columns represents 1000 steps,
Rows represent learning rates, ["2.5e-5", "5e-5", "1e-4", "2e-4", "4e-4"] starting from top.

**Comment by brian6091** [comment]
Interesting, are you running a constant learning rate schedule without warmup?

**Reply by cloneofsimo** [reply]
Yes I am running constant learning rate!

**Reply by brian6091** [reply]
Who is the target person? Do you have a link to an image of what they are supposed to look like?

**Reply by cloneofsimo** [reply]
She is game character from overwatch.
for the lack of a better answer : https://www.youtube.com/watch?v=9acxn7qAST4

**Reply by cian0** [reply]
based on my own testing, using the linear decay starting from 1e-4 works for real people but when I use it with other concepts (e.g. cyberpunk, steampunk etc) it couldn't capture it well, but the token alone would show good likeness

**Reply by sitatec** [reply]
@cian0 were you able to find a config that work well for human faces with concepts like cyberpunk or steampunk?

**Reply by cian0** [reply]
@sitatec honestly the results are varying, sometimes 8k steps work with one face, then wouldn't even capture another, so I'm assuming that lora might work mostly on faces that are already similar to what's already inside the model. I'm reverting to shiv's for now

**Reply by brian6091** [reply]
@cian0 , would you happen to have a small dataset for a face where LoRA isn't doing well? I had some difficulty with a person that I made sure wasn't in the SD model, and I wonder if it's my source data.

**Reply by cian0** [reply]
@brian6091 sure, though i couldn't share them publicly, you're in Dreambooth discord as well right? I'll add you there and send some test dataset when i get back to my workstation

**Reply by brian6091** [reply]
@cian0 yeah, on the Dreambooth discord. Thanks!

**Comment by cloneofsimo** [comment]
# Pivotal Tuning

Rank1, fine-tunes FF as well.
Every image represents 500 steps, inversion happens until 2500 steps, fine-tuning kicks in after. 

### 2e-6
![2e-6-krk-pt](https://user-images.githubusercontent.com/35953539/208834635-02c0cb85-0a96-4eff-8da9-5c1c7748f740.jpg)
### 5e-6
![5e-6-krk-pt](https://user-images.githubusercontent.com/35953539/208829359-2fafb080-5743-4221-bd50-49782a861141.jpg)
### 1e-5
![1e-5-krk-pt](https://user-images.githubusercontent.com/35953539/208829368-cca58ffd-5504-4767-b0b6-c121a3c8a3f4.jpg)
### 2e-5
![2e-5-krk-pt](https://user-images.githubusercontent.com/35953539/208829383-6793aba3-bea3-4016-be0d-89c2c19ea7a5.jpg)

**Reply by cian0** [reply]
So 6k @ 2e-5 seems the best?

**Comment by cloneofsimo** [comment]
More results on Inversion:
So technically not related to LoRA, but...
# lr : 3e-4
![ti-iter](https://user-images.githubusercontent.com/35953539/208857043-9fdb77b3-95c8-4e6d-8f35-9806659e54e4.jpg)
# 1r : 1e-3
![image](https://user-images.githubusercontent.com/35953539/208860476-4daf1314-b0c0-4d4e-9c3b-bc5e4d469797.png)

**Comment by brian6091** [comment]
Going off the rails at 1e-3!

**Reply by ExponentialML** [reply]
Looks like a lower learning rate benefits from using pivotal tuning.
